src/api/drone.rs

```diff
@@ -0,0 +1,23 @@
+use crate::types::topology::routes::Path;
+use std::thread;
+use crossbeam_channel::Receiver;
+use crossbeam_channel::Sender; //still not sure about the types, didn't experiment much
+
+pub fn
+
+new_drone(drone_id: u8, _neighbors_id: Vec<u8>, chnl_recv: Vec<Receiver<Packet>>,
+                 chnl_send: Vec<Sender<Packet>>)  {
+    thread::spawn(move || {
+        let _id = drone_id;
+        let receivers = chnl_recv;
+        let _senders = chnl_send;
+
+        for i in &receivers {
+            i.recv().unwrap();
+        } //This would listen to all the channel for which this drone is a receiver
+    });
+}
```

Comment:
I'm very against the idea to add code like this to the common code. This is code that should be defined by the group not by the class. 

We should focus on struct, API and the like. This code in particular touch a portion of the code-base that probably every group would want to do in different way.

Reply:
Yea I asked him to add it, but probably tomorrow we'll go back, because it was just a sketch and shouldn't have put it there already

Reply:
But since all the drones will need to be compatible with the same initializer, at least a common idea on what to do is needed.

Reply:
Ok i understand what you are coming from. Than that is fine but it shouldn't be in this PR. Also i think that ypu just need to create the struct the initializer should be the one externally to create a thread and run this drone on it.